src/manifest/factory.ts

```diff
@@ -55,120 +49,15 @@ export default function createManifest(
   manifestObject : IParsedManifest,
   externalTextTracks : ISupplementaryTextTrack|ISupplementaryTextTrack[],
   externalImageTracks : ISupplementaryImageTrack|ISupplementaryImageTrack[],
-  warning$ : Subject<Error|ICustomError>
+  warning$ : Subject<Error|ICustomError>,
+  customRepresentationFilter? : CustomRepresentationFilter
 ) : Manifest {
-  manifestObject.periods = (manifestObject.periods).map((period) => {
-    Object.keys(period.adaptations).forEach((type) => {
-      const adaptationsForType = period.adaptations[type];
-      if (!adaptationsForType) {
-        delete period.adaptations[type];
-      } else {
-        const checkedAdaptations = checkAdaptations(adaptationsForType, warning$);
-        if (!checkedAdaptations.length) {
-          delete period.adaptations[type];
-        } else {
-          period.adaptations[type] = checkedAdaptations;
-        }
-      }
-    });
-
-    if (
-      !period.adaptations.video &&
-      !period.adaptations.audio
-    ) {
-      throw new MediaError("MANIFEST_PARSE_ERROR", null, true);
-    }
-
-    return period;
-  });
-
-  const manifest = new Manifest(manifestObject);
+  const manifest = new Manifest(
+    manifestObject,
+    warning$,
+    customRepresentationFilter
+  );
   manifest.addSupplementaryTextAdaptations(externalTextTracks);
   manifest.addSupplementaryImageAdaptations(externalImageTracks);
   return manifest;
```

Comment:
IMO, ``src/manifest/factory.ts`` can be removed here. Let's the pipeline add the tracks.

Reply:
This means that the supplementary tracks wont go through the filter, is that what we want?

Reply:
Also, I've been thinking... Wouldn't it be in fact better if the supplementary tracks were directly added to Manifest at instanciation?

Something like:
```ts
new Manifest(manifestObject, warning$, {
  supplementaryTextTracks,
  supplementaryTextTracks,
  representationFilter,
});
```

looks nicer no?